### PR TITLE
[Docs][KubeRay] Add doc for kubectl plugin

### LINKED ISSUE #47849: [KubeRay] Add doc for kubectl plugin
### Description

Add doc for KubeRay [kubectl-plugin](https://github.com/ray-project/kuberay/tree/df5577fe1b3f537f36dbda3870b4743edd2f11bb/kubectl-plugin) for KubeRay v1.2.2 release.

### Link

_No response_

### PR DESCRIPTION
**Pull request by MortalHappiness**
## Why are these changes needed?

See the description in the corresponding issue for details.

## Related issue number

Closes: ray-project/ray#47849

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [x] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+# Use Kubectl Plugin (Alpha Feature)
+
+Today it is incredibly challenging for data scientists and AI researchers unfamiliar with Kubernetes to start using Ray on Kubernetes. However, running Ray on Kubernetes is advantageous or necessary in certain environments.
+
+Therefore, starting from KubeRay v1.2.2, we have introduced a `kubectl ray` plugin to simplify the process of running Ray on Kubernetes. The plugin is designed to help users who are not familiar with Kubernetes to run Ray on Kubernetes with ease.
+
+## Installation
+
+Please follow the instructions in the [KubeRay Kubectl Plugin](https://github.com/ray-project/kuberay/tree/master/kubectl-plugin) to install the plugin.
```
Comment (andrewsykim) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:11]: We bundle the plugin binary into the KubeRay release now right? can we just download the bianry from the release?

Comment (andrewsykim) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:11]: Something like:
```
curl -LO https://github.com/ray-project/kuberay/releases/download/v1.2.2/kubectl-ray_v1.2.2_linux_amd64
cp kubectl-ray_v1.2.2_linux_amd64 /usr/local/bin/kubectl-ray 
```

Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:11]: Issue created: https://github.com/ray-project/kuberay/issues/2449

Comment (andrewsykim) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:11]: Thanks I think we can directly add this part into this page too:
```
curl -LO https://github.com/ray-project/kuberay/releases/download/v1.2.2/kubectl-ray_v1.2.2_linux_amd64.tar.gz
tar -xvf kubectl-ray_v1.2.2_linux_amd64.tar.gz
cp kubectl-ray ~/.local/bin
```

Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:11]: Done

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+And a file named `ray-job.usermode.yaml`
+
+```yaml
+apiVersion: ray.io/v1
+kind: RayJob
+metadata:
+  name: rayjob-usermode
+spec:
+  submissionMode: UserMode
```
Comment (andrewsykim) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:119]: Given we're about to change this for v1.3, should we remove any examples of `kubctl ray job submit`? I'm worried users will try with UserMode with 1.2 and then we'll break them in 1.3

Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md]: Done in https://github.com/ray-project/ray/pull/47851/commits/c87ec4abd47f7fd52c43b688556d854f3664e886. `kubectl ray job submit` example is completely removed.

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
+(kubectl-plugin)=
+
+# Use Kubectl Plugin (Alpha Feature)
+
+Today it is incredibly challenging for data scientists and AI researchers unfamiliar with Kubernetes to start using Ray on Kubernetes. However, running Ray on Kubernetes is advantageous or necessary in certain environments.
```
Comment (andrewsykim) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:5]: I think we can remove this sentence. This was important context for why we built it, but we don't need to include it in the guide. 
```
Starting from KubeRay v1.2.2, we have introduced a `kubectl ray` plugin to simplify common workflows when deploying Ray on Kubernetes. The plugin is designed to help users who are not familiar with Kubernetes to run Ray on Kubernetes with ease.
```

**Review by andrewsykim — APPROVED** [review]

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md (hunk 3 above)
Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md]: Done in https://github.com/ray-project/ray/pull/47851/commits/e802c4ba0bb1ec5b5624ef664180b73e5e72426b

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+Forwarding from [::1]:10001 -> 10001
+```
+
+### Get Ray cluster logs
+
+```shell
+$ kubectl ray log ray-cluster-kuberay
+
+No output directory specified, creating dir under current directory using cluster name.
+Downloading file ./ for Ray Head ray-cluster-kuberay-head-qkrv8
```
Comment (chiayi) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:61]: Minor change with https://github.com/ray-project/kuberay/pull/2442. The output is now just the ray node name. Ex:
```
$ kubectl ray logs rayjob-sample-raycluster-kfhl6
No output directory specified, creating dir under current directory using cluster name.
Command set to retrieve both head and worker node logs.
Downloading log for Ray Node rayjob-sample-raycluster-kfhl6-head-87xpb
Downloading log for Ray Node rayjob-sample-raycluster-kfhl6-small-group-worker-54qfm
```

Comment (MortalHappiness) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md]: Addressed in https://github.com/ray-project/ray/pull/47851/commits/91a1ee2560411a61f0ad6219dd3bb6480d7df085

**Comment by andrewsykim** [comment]
@angelinalg @kevin85421 can you help review please?

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+After installing the plugin, you can use `kubectl ray --help` to see the available commands and options.
+
+## Example
+
+Here assume you have a Ray cluster running on Kubernetes. Please see {ref}`RayCluster Quickstart <kuberay-raycluster-quickstart>` if you don't have a Ray cluster running on Kubernetes.
+
+### Get all Ray clusters
+
+```shell
+$ kubectl ray cluster get
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:45]: Maybe remove this section for now. If I remember correctly, we decided to change to `kubectl ray get [cluster | job | service]` to make the syntax closer to `kubectl get [pods, service, ...]`. We can add it back after we change the syntax.

**Review by kevin85421 — APPROVED** [review]

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md (hunk 5 above)
Comment (kevin85421) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:45]: cc @andrewsykim

**Comment by kevin85421** [comment]
I have already pinged our doc team to review this PR.

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
+(kubectl-plugin)=
+
+# Use Kubectl Plugin (Alpha Feature)
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:3]:
```suggestion
# Use Kubectl plugin (alpha)
```

Comment (angelinalg) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:3]: Sentence case for titles.

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+
+Starting from KubeRay v1.2.2, we have introduced a `kubectl ray` plugin to simplify common workflows when deploying Ray on Kubernetes. The plugin is designed to help users who are not familiar with Kubernetes to run Ray on Kubernetes with ease.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:5]:
```suggestion
Starting from KubeRay v1.2.2, you can use the `kubectl ray` plugin to simplify common workflows when deploying Ray on Kubernetes. If you aren't familiar with Kubernetes, this plugin simplifies running Ray on Kubernetes.
```

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+
+## Installation
+
+Please follow the instructions in the [KubeRay Kubectl Plugin](https://github.com/ray-project/kuberay/tree/master/kubectl-plugin) to install the plugin.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:9]:
```suggestion
See [KubeRay Kubectl Plugin](https://github.com/ray-project/kuberay/tree/master/kubectl-plugin) to install the plugin.
```

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+
+You can install the Kuberay kubectl plugin using one of the following methods:
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:11]:
```suggestion
Install the Kuberay kubectl plugin using one of the following methods:

-  Use the Krew kubectl plugin manager (recommended)
-  Download from GitHub releases

```

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+### Install using Krew kubectl plugin manager (Recommended)
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:13]:
```suggestion
### Install using the Krew kubectl plugin manager (recommended)
```

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+
+Replace `~/.local/bin` with the directory in your `PATH`.
+
+## Usage
+
+After installing the plugin, you can use `kubectl ray --help` to see the available commands and options.
+
+## Example
+
+Here assume you have a Ray cluster running on Kubernetes. Please see {ref}`RayCluster Quickstart <kuberay-raycluster-quickstart>` if you don't have a Ray cluster running on Kubernetes.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:40]:
```suggestion
This example assumes you have a Ray cluster running on Kubernetes. See {ref}`RayCluster Quickstart <kuberay-raycluster-quickstart>` if you don't have a Ray cluster running on Kubernetes.
```

**Review by angelinalg — COMMENTED** [review]
Just some style nits. Thanks!

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 12]
```diff
@@ -0,0 +1,65 @@
+(kubectl-plugin)=
+
+# Use Kubectl Plugin (alpha)
+
+Starting from KubeRay v1.2.2, you can use the `kubectl ray` plugin to simplify common workflows when deploying Ray on Kubernetes. If you aren't familiar with Kubernetes, this plugin simplifies running Ray on Kubernetes.
+
+## Installation
+
+See [KubeRay Kubectl Plugin](https://github.com/ray-project/kuberay/tree/master/kubectl-plugin) to install the plugin.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:9]:
```suggestion
See [KubeRay kubectl Plugin](https://github.com/ray-project/kuberay/tree/master/kubectl-plugin) to install the plugin.
```

File: doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+Install the Kuberay kubectl plugin using one of the following methods:
+
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:12]:
```suggestion

- Install using Krew `kubectl` (recommended)
- Download from GitHub releases

```

Comment (angelinalg) [reply doc/source/cluster/kubernetes/user-guides/kubectl-plugin.md:12]: I meant to add this comment. If you can include this, that would be nice. Thanks!

**Review by angelinalg — APPROVED** [review]

**Pull request merged**